packages/components/src/utils/importDataHelpers.ts

```diff
@@ -0,0 +1,30 @@
+import { verifyVC, decodeEnvelopedVC } from '@mock-app/services';
+import { VerifiableCredential } from '@vckit/core-types';
+import JSONPointer from 'jsonpointer';
+
+export const processVerifiableCredentialData = async (data: any, credentialPath?: string) => {
+  try {
+    let vc: VerifiableCredential = data as VerifiableCredential;
+    if (credentialPath) {
+      vc = JSONPointer.get(data, credentialPath);
+    }
+
+    const result = await verifyVC(vc);
+    if (!result.verified) {
+      throw new Error('Invalid Verifiable Credential!');
+    }
+    const decodedEnvelopedVC = decodeEnvelopedVC(vc);
+    if (decodedEnvelopedVC) {
+      const vcData = { vc, decodedEnvelopedVC };
+      if (credentialPath) {
+        JSONPointer.set(data, credentialPath, vcData);
+      } else {
+        data = vcData;
+      }
+    }
+
+    return data;
+  } catch (error: any) {
+    throw new Error(error);
+  }
```

Comment:
I think the exception thrown here should provide a more descriptive error message to make debugging easier and to better understand the issue.

Reply:
If we keep it the same, sometimes this function will throw an error like:
`Uncaught Error: Error: Invalid Verifiable Credential!`

Reply:
Hi @huynguyen-hl, how do you propose we address this? Are you thinking of having a generic error message instead of propagating the error message thrown or modifying the error messages thrown upstream?

Reply:
This code can be improved for better error handling. Wrapping the error in a new Error object loses the original stack trace, which can make debugging more challenging. Instead, you can throw the original error or re-throw it to preserve its information:

```ts
} catch (error) {
    throw error; // re-throw the original error to preserve stack trace
}
```

Alternatively, if you need to add context to the error, you can include additional information without losing the original error’s details:

```ts
} catch (error: any) {
    throw new Error("Custom error message", { cause: error });
}
```

Reply:
Okay. The issue, I believe, is that the stack trace is not preserved once the error is thrown.

In your first example, we might as well remove the try-catch block, right?

In the second example, yes, we preserve the stack trace but lose any well-crafted error messages in exchange for the custom message.

How do you think we could improve this, or is the custom message with the stack trace preserved sufficient?

Reply:
From my perspective, if we’re not adding more details to the error in the catch block, we might consider removing the try-catch block altogether.

Reply:
Thanks @huynguyen-hl and @ashleythedeveloper , I updated it